library/src/main/java/com/novoda/downloadmanager/lib/DownloadProvider.java

```diff
@@ -907,14 +907,15 @@ private SqlSelection getWhereClause(final Uri uri, final String where, final Str
             selection.appendClause(Downloads.Impl._ID + " = ?", getDownloadIdFromUri(uri));
         }
         if (uriMatch == BATCHES_ID) {
-            selection.appendClause(Downloads.Impl.Batches._ID + " = ?", ContentUris.parseId(uri));
+            selection.appendClause(Downloads.Impl.Batches._ID + " = ?", String.valueOf(ContentUris.parseId(uri)));
```

Comment:
Changed now: https://github.com/novoda/download-manager/commit/d1472e5d719893d361e1f1631e522d1eb141e8a4
And I've already changed `appendClause()`: https://github.com/novoda/download-manager/pull/63/files#diff-d18483ff140c15dd49ed3d88822df403L233